### DISCUSSION TITLE
Bulk install of Thorium Reader

### DISCUSSION THREAD

**Discussion by llemeurfr** [edrlab/thorium-reader#1642] [Ideas]
From a user: 
"i work in a school as an IT administrator and one of the teacher in his class wants to have thorium on all of his computer in his class. i have tried looking for hours but could not find the way to deploy it to multiple computers. i have also used some scripts to run installer but no luck. can you please suggest me the way to deploy it to multiple users."

**Comment by mutterback** [comment]
Hi, happy to share some of our experience. 

The most automated way, with many strings attached, is to use Microsoft InTune or another device management application. InTune works on Windows, Android and iOS, lots of other 3rd party solutions for both OSs. Google, Samsung, HP all have tools for this. These solutions are not free and the learning curve is steep, but are designed to do what the IT administrator is looking for. They allow you to push apps from a central console to all of your devices. These solutions require that you put the devices under management, and apply *a lot* of detailed policy settings which range from other apps users can install, downloads, and even web site use. InTune and Google Mobile Device Management can also create managed app store accounts, which we find to be a major pain on shared school devices. Windows and Google Play assume you are an individual using a device with a personal email, mobile, and credit card. 

We also see schools constantly pass up the opportunity to work with a device supplier to set up their devices with a profile when they purchase them. This would mean their devices would come with the apps, policies, files even user account set up without needing to unbox and set up every device 1 by 1. If you ask, a device supplier should be willing to do this for an order of 20 devices or more. Schools just don't ask and are leaving a lot on the table. Same is true if you are going with Office/Windows 365 or Google Apps for education. Push your VAR to get device management with what you're paying for and support you to implement it. There are also a lot of educational discounts we see schools pass up because they just don't ask.

I believe Daniel was looking into code signing + installer packaging. I'm not sure if he did that for Thorium. But, that could make the install process a bit easier. That is still 1 device at a time, but wouldn't require device management and would work without needing to download the app for every device. Should also make scripting from a file share possible, though I haven't tried it.

**Comment by atomotic** [comment]
maybe build as [portable target](https://www.electron.build/configuration/win). no installation needed, could be enough to copy a directory to all computers.
but not sure if this registers file extensions to be opened with thorium

**Comment by sukiletxe** [comment]
I believe as a result of this being in Microsoft Store it is also available in winget, so one can install it with winget (command line), with `winget install thorium` or something similar.

**Reply by danielweck** [reply]
Warning: this WinGet distribution is not owned/maintained by EDRLab:

https://winget.run/pkg/EDRLab/Thorium

**Reply by danielweck** [reply]
Warning: this WinGet distribution is not owned/maintained by EDRLab:
https://github.com/SpecterShell/winget-pkgs/blob/master/manifests/e/EDRLab/Thorium/3.3.0/EDRLab.Thorium.installer.yaml

**Reply by danielweck** [reply]
Note: for MacOS, there is also a Homebrew distribution not maintained/owned by EDRLab:
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/t/thorium.rb

**Comment by tipietil** [comment]
I'm also working for IT and would like to have installer for all users on the computer, for multiple computers, current installer makes it for current user only, and then gets blocked by applocker. Additional problem is that blocked .exe is not signed, so you can't do publisher rule for that .exe. File hash rule would be bad because that changes when version changes. Path rule makes a security hole which is inadvisable.

**Comment by gustavenne** [comment]
I'm deploying to a small fleet using a simple silent install script with command Thorium.Setup.2.3.0.exe /S .
Is there a way to force a specific language during installation? I tried the arguments /LANG=1036 or /LANG=fr-fr and others but it will always install in english no matter what.

**Reply by llemeurfr** [reply]
Thorium is not installed in a unique language: all languages are present in the app, English is presented first as it is the "lingua franca" these days. I imagine that you'd like to force the langage selected on te first launch.

**Reply by gustavenne** [reply]
Exactly, we would like the application to be in french at first launch if possible by any tweak. I've tried to look in the code where the default "first launch" display language is set by I wouldn't be competent enough to edit and rebuild an executable. Thanks!

**Reply by llemeurfr** [reply]
and if we manage to launch the app by default (first time) in the language of the OS (if there is a match with a supported language), with an english fallback?

**Reply by gustavenne** [reply]
That should work, in our case it's a bit of a complicated deployment method. OS is installed in English first for ease of path names and GPO. Then the French Canadian language pack is applied and we basically set everything to french Canada, systemUI, Input method, Locale. What could be a quick workaround would be to know in which file the chosen language gets stored if selected after first launch. We could then deploy a simple invisible script to run post-install to overwrite that file, eg replace en-US to fr-fr or whatever the changed value is for french. It wouldn't be a clean workaround but if it works, that could be enough until something like "set default language to OS language" gets added. Thanks a lot!

**Reply by llemeurfr** [reply]
There is no file implied outside the executable. The localizations are handled inside Thorium. We could only program a "test" of the system language on the first launch (after that, it is user's choice that sets the UI language).

**Reply by gustavenne** [reply]
I think I'll familiarise myself with the code a bit and try to contribute best I can. If it gets decided to implement this "system language on the first launch" in the future, I'll try to contribute so that system languages like french Canada (fr-CA) get set to the current French json file in place. There would be no point in having multiple french variant for the App's UI in my opinion so most french country systems such as fr-CA, fr-BE, fr-CH, fr-CD and others should probably all point to the current french json already in place. Thank you for this great project and support

**Comment by danielweck** [comment]
Related issue: https://github.com/edrlab/thorium-reader/issues/3371

**Reply by llemeurfr** [reply]
In this issue, there is a mention of "/S /AllUsers" arguments. The first argument means "Silent", the second "available for all users".

**Comment by danielweck** [comment]
See https://github.com/edrlab/thorium-reader/issues/3327

Official documentation: https://learn.microsoft.com/en-us/microsoft-365/solutions/apps-guide-overview?view=o365-worldwide

**Comment by danielweck** [comment]
related issue https://github.com/edrlab/thorium-reader/issues/3586
